src/helpers/parser/listItemParser.js

```diff
@@ -495,7 +495,7 @@ export const parseListItemsFromQuery = (query, data, titleDetail, options = {})
     case QUERY_TYPES.VOLUNTEER.PROFILE:
       return parseVolunteers(data, query, skipLastDivider);
     case QUERY_TYPES.VOUCHERS:
-      return parseVouchers(data[QUERY_TYPES.GENERIC_ITEMS], skipLastDivider);
+      return parseVouchers(data[QUERY_TYPES.GENERIC_ITEMS] || data, skipLastDivider);
```

Comment:
is `|| data` really needed her? or could it be passed as `{ [QUERY_TYPES.GENERIC_ITEMS]: data }` in here, to not need the condition?

Reply:
You need it here. If the data is not in the `genericItems` object, I think `{ [QUERY_TYPES.GENERIC_ITEMS]: data }` will give an error.

Reply:
But then what about the data inside the genericItem object? Then won't genericItem.genericItem = data?

Reply:
my suggestion should only be applied where data is passed directly without `genericItem:` to ensure, that this key exists in any case.

Reply:
Aha. This is what you are talking about. Yes. We can delete the logic in `listItemParser`.

```
  const voucherListItems = useMemo(() => {
    return parseListItemsFromQuery(
      QUERY_TYPES.VOUCHERS,
      { [QUERY_TYPES.GENERIC_ITEMS]: vouchers },
      undefined,
      {
        withDate: false
      }
    );
  }, [vouchers]);
```

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/pull/1072/commits/b7eead536633a9e4c6b0a47ace778ba0ece94787